cmd/promtool/main.go

```diff
@@ -648,16 +646,25 @@ func QueryLabels(url *url.URL, name string, start, end string, p printer) int {
 	for _, v := range warn {
 		fmt.Fprintln(os.Stderr, "query warning:", v)
 	}
-
-	if err != nil {
-		fmt.Fprintln(os.Stderr, "query error:", err)
+	if apiErr := handleAPIError(err); apiErr != nil {
 		return 1
 	}
```

Comment:
```suggestion
	if err != nil {
		return handleAPIError(err)
	}
```

This will make the function a bit easier, also, it will match the patter we use for  invalidParamError

Reply:
Makes sense! I thought about possibility of clearing the `err` in some cases, but that's rather rare thing.